sql/truncate.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: XisiHuang (cockhuangxh@163.com)
+
+package sql
+
+import (
+	"github.com/cockroachdb/cockroach/client"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/sql/parser"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+// Truncate deletes all rows from a table.
+func (p *planner) Truncate(n *parser.Truncate) (planNode, error) {
+	b := client.Batch{}
```

Comment:
Tobias,

We need to figure out the transaction story here. At least my thinking is
that we pass in a transaction object into all these sql methods. if you
take a look pete has marked commit() and
Rollback() as unimplemented.

On Mon, Aug 3, 2015 at 8:30 AM Tobias Schottdorf <notifications@github.com>
wrote:

> In sql/truncate.go
> <https://github.com/cockroachdb/cockroach/pull/1906#discussion_r36080184>:
>
> > +// for names of contributors.
> > +//
> > +// Author: XisiHuang (cockhuangxh@163.com)
> > +
> > +package sql
> > +
> > +import (
> > +	"github.com/cockroachdb/cockroach/client"
> > +	"github.com/cockroachdb/cockroach/proto"
> > +	"github.com/cockroachdb/cockroach/sql/parser"
> > +	"github.com/cockroachdb/cockroach/util/log"
> > +)
> > +
> > +// Truncate deletes all rows from a table.
> > +func (p *planner) Truncate(n *parser.Truncate) (planNode, error) {
> > +	b := client.Batch{}
>
> is the plan that all those methods get a client object passed down to them
> (which would always be transactional)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1906/files#r36080184>.
>